sdk/dotnet/Pulumi/Deployment/Outputs.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016-2022, Pulumi Corporation
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// A type-safe wrapper around <see cref="IDictionary{String, Object}"/> for generating stack outputs from created resources. 
+    /// </summary>
+    public class Outputs : IEnumerable<KeyValuePair<string, object?>>
```

Comment:
Can we make it implement `IDictionary` and reuse the existing `RynAsync` overloads? What am I missing?

Reply:
I tried implementing `IDictionary` but it forces `Outputs` to implement `Add(string, object)` and untyped set indexer which is not exactly the idea here. Also tried implementing `IReadOnlyDictionary` but it could not be implicitly converted to `IDictionary` when using the existing `RunAsync` methods.

Reply:
What about an implicit conversion operator to IDictionary?